cmd/juju/application/bundle_test.go

```diff
@@ -174,6 +174,33 @@ func (s *BundleDeployCharmStoreSuite) TestDeployBundleStorage(c *gc.C) {
 	})
 }
 
+func (s *BundleDeployCharmStoreSuite) TestDeployBundleDevices(c *gc.C) {
+	_, minerCharm := testcharms.UploadCharm(c, s.client, "xenial/miner-42", "miner-devices")
+	_, wpch := testcharms.UploadCharm(c, s.client, "xenial/wordpress-47", "wordpress")
+	testcharms.UploadBundle(c, s.client, "bundle/wordpress-dashboard-with-miner-backend", "wordpress-dashboard-with-miner-backend")
+	err := runDeploy(
+		c, "bundle/wordpress-dashboard-with-miner-backend",
+		"--storage", "miner:bitcoinminer=10,nvidia.com/gpu", // override bitcoinminer
```

Comment:
Is this passing an nvidia.com/gpu to '--storage' ? certainly that isn't the right CLI flag to be using for all devices, is it?

Reply:
Thanks for reviewing, @jameinel . Currently, this PR is still WIP and I am doing cleanup and fixing failed tests now. I will ask for your help to take a look again soon when it's ready. 
Thanks.